src/psr4/Bridge.php

```diff
@@ -754,13 +754,21 @@ public function remove_filter( $hook, $mvc_handler, $priority = 10 )
      * Returns class method call mapped to a mvc engine method.
      * @since 1.0.3
      *
+     * @param string|array $call
+     *
      * @return string
      */
     private function get_mapped_mvc_call( $call, $return = false )
     {
-        return ( preg_match( '/[vV]iew\@/', $call ) ? '_v_' : '_c_' )
-            . ( $return ? 'return_' : 'void_' )
-            . $call;
+        if ( is_array( $call ) ) {
+            $class_prefix = $this->config->get( 'namespace' ) . '\\Controllers\\';
+            $call = str_replace( $class_prefix, '', implode( '@', $call ) );
+        }
+
+        $type_prefix = stripos( $call, 'view@' ) !== false ? '_v_' : '_c_';
+        $return_prefix = $return ? 'return_' : 'void_';
+
+        return $type_prefix . $return_prefix . $call;
```

Comment:
To optimize performance and improve memory allocation, we want to avoid instantiating variables unless is highly necessary. Please do the concatenation directly, the same as the original code that was refactored, and remove all the variables that are only used once.

Reply:
Rolled back.